src/include/mpir_contextid.h

```diff
@@ -8,9 +8,19 @@
 #ifndef MPIR_CONTEXTID_H_INCLUDED
 #define MPIR_CONTEXTID_H_INCLUDED
 
+#ifdef HAVE_EXTENDED_CONTEXT_BITS
+#define MPIR_CONTEXT_ID_T_DATATYPE MPI_UINT32_T
+typedef uint32_t MPIR_Context_id_t;
+#define MPIR_INVALID_CONTEXT_ID ((MPIR_Context_id_t)0xffffffff)
+#define MPIR_CONTEXT_ID_BITS 20
+#define CONTEXT_ID_FMT PRIu32
```

Comment:
Seems `MPIR_Context_id_t` just need be an int type that can accommodate maximum context id value, so we could use `uint32_t` for both cases, right? If so, then I think we should prefer that as it simplifies the handling.

Reply:
It would affect the size of the ch3 pkt headers, probably unless we used a bitfield. We've gotten dinged by derivatives for messing with the pkt header size in the past, so I was being conservative here.

Reply:
I see. Could you add a comment refer to this caution?

Reply:
Sure, added in this version.